query-node/mappings/mappingsContent.ts

```diff
@@ -0,0 +1,464 @@
+  VideoId,
+  VideoCreationParameters<ContentParameters>,
+  */
+
+  const protobufContent = readProtobuf(ProtobufEntity.Video, (event.params[3] as any).meta) // TODO: get rid of `any` typecast
+
+  const channel = new Video({
+    id: event.params[1].toString(), // ChannelId
+    isCensored: false,
+    channel: event.params[1],
+    happenedIn: convertblockNumberToBlock(event.blockNumber),
+    ...protobufContent.toObject()
+  })
+
+  await db.save<Video>(channel)
+}
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export async function content_VideoUpdated(
+  db: DatabaseManager,
+  event: SubstrateEvent
+) {
+  /* event arguments
+  ContentActor,
+  VideoId,
+  VideoUpdateParameters<ContentParameters>,
+  */
+  const videoId = event.params[1].toString()
+  const video = await db.get(Video, { where: { id: videoId } })
+
+  const protobufContent = readProtobuf(ProtobufEntity.Video, (event.params[2] as any).meta) // TODO: get rid of `any` typecast
+
+  const updatedVideo = new Video({
+    // TODO: check that this kind of new updated channel construction is valid or it should rather be edited as `channel.myProperty = something`
+    ...video,
+    ...protobufContent.toObject()
+  })
+
+  await db.save<Video>(video)
+}
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export async function content_VideoDeleted(
+  db: DatabaseManager,
+  event: SubstrateEvent
+) {
+  /* event arguments
+  ContentActor,
+  VideoCategoryId,
+  */
+
+  const videoId = event.params[1].toString()
+  const video = await db.get(Video, { where: { id: videoId } })
+
+  await db.remove<Video>(video)
+}
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export async function content_VideoCensored(
+  db: DatabaseManager,
+  event: SubstrateEvent
+) {
+  /* event arguments
+  ContentActor,
+  VideoId,
+  Vec<u8>
+  */
+
+  const videoId = event.params[1].toString()
+  const video = await db.get(Video, { where: { id: videoId } })
+
+  video.isCensored = true;
+
+  await db.save<Video>(video)
+}
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export async function content_VideoUncensored(
+  db: DatabaseManager,
+  event: SubstrateEvent
+) {
+  /* event arguments
+  ContentActor,
+  VideoId,
+  Vec<u8>
+  */
+
+  const channelId = event.params[1].toString()
+  const video = await db.get(Video, { where: { id: videoId } })
+
+  video.isCensored = false;
+
+  await db.save<Video>(video)
+}
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export async function content_FeaturedVideosSet(
+  db: DatabaseManager,
+  event: SubstrateEvent
+) {
+  /* event arguments
+  ContentActor,
+  Vec<VideoId>,
+  */
+
+  const videoIds = event.params[1]
+
+  for (let videoId in videoIds) {
+    const video = await db.get(Video, { where: { id: videoId } })
+
+    video.isFeatured = true;
+
+    await db.save<Video>(video)
+  }
+
+  // TODO: remove featured flag from previously featured videos
```

Comment:
Yes. Try to do the update in the most efficient way by reducing number of db updates, by getting array of current featured videos, and intersecting with array of new set of featured videos to get two array of videos to unset and videos to set..

Reply:
Diff sets are now calculated, and only changed videos in the featured video list will get updated.